### PR TITLE
Improve BeEquivalentTo error messages

### PR DESCRIPTION
**Pull request by ronaldkroon**
* Changed sentence to avoid confusion of _actual_ document after "Expected"
* Mention actual type and length differences in message

### CODE REVIEW

File: Src/FluentAssertions.Json/JTokenDifferentiator.cs [hunk 1]
```diff
@@ -166,19 +166,71 @@ private static Difference CompareValues(JValue actual, JValue expected, JPath pa
...
+                    return "a timespan";
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(jTokenType), jTokenType, null);
+            }
+        }
     }
 
     internal class Difference
     {
+        public Difference(DifferenceKind kind, JPath path, object actual, object expected) : this(kind, path)
```
Comment (dennisdoomen) [inline Src/FluentAssertions.Json/JTokenDifferentiator.cs]: Shouldn't `actual` and `expected` be of type `string`?

File: Tests/FluentAssertions.Json.Shared.Specs/JTokenAssertionsSpecs.cs [hunk 2]
```diff
@@ -65,103 +65,114 @@ public void When_both_objects_are_the_same_or_equal_BeEquivalentTo_should_succee
...
-                string expectedDifference = testCase.Item3;
+                    "has a different value at $.id"
+                };
+            }
+        }
 
-                var actual = (actualJson != null) ? JToken.Parse(actualJson) : null;
-                var expected = (expectedJson != null) ? JToken.Parse(expectedJson) : null;
+        [Theory, MemberData(nameof(FailingBeEquivalentCases))]
+        public void When_objects_differ_BeEquivalentTo_should_fail(string actualJson, string expectedJson,
```
Comment (dennisdoomen) [inline Tests/FluentAssertions.Json.Shared.Specs/JTokenAssertionsSpecs.cs]: Rather then referring to the name of the method, I suggest to rewrite it as `When_json_fragements_are_not_equivalent_it_should_throw`

File: Src/FluentAssertions.Json/JTokenDifferentiator.cs (hunk 1 above)
Comment (ronaldkroon) [reply Src/FluentAssertions.Json/JTokenDifferentiator.cs]: I'm also passing `int`s. This way I don't have to call `ToString`.

**Comment by dennisdoomen** [comment]
Released as 5.4.0

**Pull request merged**